From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aditya Prayoga <aditya@kobol.io>
Date: Tue, 15 Sep 2020 20:04:22 +0700
Subject: arm64: dts: rockchip: helios64: add vdd_center and vdd_log

RK808 DCDC_REG1 vdd_center and the PWM2 regulator vdd_log, part of
the Armbian CPU stability work.

Upstream: none.

Split out of add-board-helios64.patch (Aditya Prayoga, rebased onto
mainline by rpardini for 6.3); the DTB content is unchanged.

Signed-off-by: Aditya Prayoga <aditya@kobol.io>
Assisted-by: LLM
Signed-off-by: Igor Velkov <iav@iav.lv>
---
 .../boot/dts/rockchip/rk3399-kobol-helios64.dts    | 32 ++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-kobol-helios64.dts b/arch/arm64/boot/dts/rockchip/rk3399-kobol-helios64.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-kobol-helios64.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-kobol-helios64.dts
@@ -393,6 +393,21 @@ / {
 		regulator-max-microvolt = <12000000>;
 		vin-supply = <&vcc12v_dcin_bkup>;
 	};
+
+	vdd_log: vdd-log {
+		compatible = "pwm-regulator";
+		pwms = <&pwm2 0 25000 1>;
+		pwm-supply = <&vcc5v0_sys>;
+		regulator-name = "vdd_log";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <830000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-state-mem {
+			regulator-on-in-suspend;
+			regulator-suspend-microvolt = <900000>;
+		};
+	};
 };
 
 &cpu_b0 {
@@ -476,6 +491,19 @@ &i2c0 {
 		#clock-cells = <1>;
 
 		regulators {
+			vdd_center: DCDC_REG1 {
+				regulator-name = "vdd_center";
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1000000>;
+				regulator-ramp-delay = <6001>;
+				regulator-state-mem {
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <950000>;
+				};
+			};
+
 			vdd_cpu_l: DCDC_REG2 {
 				regulator-name = "vdd_cpu_l";
 				regulator-always-on;
@@ -816,6 +844,10 @@ &pwm1 {
 	status = "okay";
 };
 
+&pwm2 {
+	status = "okay";
+};
+
 &pwm3 {
 	status = "okay";
 };
-- 
Armbian

